### PR TITLE
Allow multiple needles for the changelog

### PR DESCRIPTION
**Pull request by sdepold**
This commit adds support for the `## Unreleased` needle.

### CODE REVIEW

File: lib/support/changelog.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ var dateFormat = require('dateformat');
 var helper = require('./misc');
 
 var changelog = module.exports = {
+  needles: [
+    '## Upcoming',
+    '## Unreleased'
```
Comment (madtrick) [inline lib/support/changelog.js]: What's the goal of using `Unreleased`?

Comment (madtrick) [inline lib/support/changelog.js]: I mean, what's the difference between `Upcoming` and `Unreleased`?

Comment (sdepold) [inline lib/support/changelog.js]: None. Just a matter of personal taste (the original keepachangelog actually promotes `Unreleased`)

File: lib/support/changelog.js [hunk 2]
```diff
@@ -30,17 +35,24 @@ var changelog = module.exports = {
   },
 
   needsUpdate: function () {
-    return this.read().indexOf('## Upcoming') > -1;
+    var changelog = this.read();
+
+    return this.needles.map(function (needle) {
+      return changelog.indexOf(needle) > -1;
+    }).indexOf(true) > -1;
```
Comment (madtrick) [inline lib/support/changelog.js]: what about rewriting this using `some`?

```js
return this.needles.some(function (needle) {
     return changelog.indexOf(needle) > -1;
});
```

[some function on mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)

File: lib/support/changelog.js (hunk 1 above)
Comment (madtrick) [inline lib/support/changelog.js]: Ah ok : )

File: lib/support/changelog.js (hunk 2 above)
Comment (sdepold) [inline lib/support/changelog.js]: Awesome! Exactly what I was wondering about!

Comment (sdepold) [inline lib/support/changelog.js]: fixed

**Pull request merged**